packages/framework/aqueduct/src/data-object-factories/pureDataObjectFactory.ts

```diff
@@ -259,8 +259,25 @@ export class PureDataObjectFactory<
 		runtime: IContainerRuntime,
 		initialState?: I["InitialState"],
 	): Promise<TObj> {
-		const context = runtime.createDetachedRootDataStore([this.type], rootDataStoreId);
-		return this.createInstanceCore(context, initialState);
+		const dataStore = (runtime as ContainerRuntime).createDataStore2([this.type]);
```

Comment:
I assume casting and createDataStore2 are for prototype only, right? I.e. long-term we should find better names and avoid any casting.

Reply:
100%, this is a prototype for now, we may make these changes directly to the IDataStore interface. I'm not sure what exact direction we should be headed in.